activation/activation_errors.go

```diff
@@ -26,3 +27,31 @@ func (e *PoetSvcUnstableError) Error() string {
 }
 
 func (e *PoetSvcUnstableError) Unwrap() error { return e.source }
+
+type PoetRegistrationMismatchError struct {
+	registrations   []string
+	configuredPoets []string
+}
+
+func (e *PoetRegistrationMismatchError) Error() string {
+	var sb strings.Builder
+	sb.WriteString("builder: none of configured poets matches the existing registrations.\n")
+	sb.WriteString("registrations:\n")
+	for _, r := range e.registrations {
+		sb.WriteString("\t")
+		sb.WriteString(r)
+		sb.WriteString("\n")
+	}
+	sb.WriteString("\n configured poets:\n")
+	for _, p := range e.configuredPoets {
+		sb.WriteString("\t")
+		sb.WriteString(p)
+		sb.WriteString("\n")
+	}
+	return sb.String()
+}
+
+func (e *PoetRegistrationMismatchError) As(target error) bool {
+	var poetRegistrationMismatchError *PoetRegistrationMismatchError
+	return errors.As(target, &poetRegistrationMismatchError)
+}
```

Comment:
You can delete this method. It isn't needed, it might even lead to an endless loop, where `errors.As` calls this method, which calls `errors.As`.